addons/account/models/account_move.py

```diff
@@ -2604,7 +2604,15 @@ def _get_quick_edit_suggestions(self):
                     self.journal_id.company_id.account_purchase_tax_id
                 )
             taxes = self.fiscal_position_id.map_tax(taxes)
-        price_untaxed = taxes.with_context(force_price_include=True).compute_all(
+
+        # When a payment term has an early payment discount and the company's epd computation is set to 'mixed', recomputing
+        # the untaxed amount should take in consideration the discount percentage otherwise we'd get a wrong value.
+        # Since in a payment term we can have multiple lines with multiple discounts, handling all cases can get
+        # complicated. For this we check that we have only one line with one discount and handle only this case.
+        discount_percentage = self.invoice_payment_term_id.line_ids.discount_percentage if len(self.invoice_payment_term_id.line_ids) == 1 else 0
```

Comment:
Couldn't we also handle the case where there are multiple payment term lines, but the discount_percentage is always the same ? 

That wouldn't solve all issues but at least some of them :thinking:

Reply:
I don't really care about those cases, they are indeed dropped in 16.2. Also, this is not blocking ; if it doesn't work, the user can put 0 in Quick Encoding and revert to "normal" encoding anyway.